source/common/http/conn_manager_impl.cc

```diff
@@ -598,6 +599,17 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // called with end_stream=true.
   maybeEndDecode(end_stream);
 
+  if (connection_manager_.config_.validateHeaders()) {
+    ENVOY_STREAM_LOG(trace, "validating header values\n{}", *this, *request_headers_);
+    if (!Http::HeaderUtility::validateHeaders(*request_headers_)) {
```

Comment:
should we bump a stat here? might be useful

Reply:
+1